src_erl/NerlnetApp/src/Bridge/nerlTests.erl

```diff
@@ -281,6 +304,56 @@ nerlworker_test_generate_data(LayersSizes, LayerTypes, NumOfSamples) -> %% Ask D
       {NerlTensor , Type} = nerlNIF:nerltensor_conversion({ErlDataTensor,erl_float} , float),
       {NerlTensor , Type , ErlDataTensor , erl_float , NumOfFeatures , NumOfLabels}.
 
+%test_count_label_nif(_Performance) -> _Performance;
+test_count_label_nif() -> 
+      ModelId  = erlang:unique_integer([positive]),
+      ModelType = "0",
+      ModelArgs = "",
+      LayersFunctionalityCodes = "1,6", 
+      LearningRate = "0.01",
+      Epochs = "50",
+      OptimizerType = "2",
+      OptimizerArgs = "",
+      LossMethod = "2",
+      DistributedSystemType = "3",
+      DistributedSystemArg = "",
+     % nerltest_print(nerl:string_format("DATA_DIM_X ~p",[?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_DIM_X])),
+      %nerltest_print(nerl:string_format("DATA_DIM_X 2 ~p",[?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_MIN_DIM_X])),
+      DimMaxDimX = ?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_DIM_X,
+      DimMinDimX = ?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_MIN_DIM_X,
+      LenDataToRand = DimMaxDimX - DimMinDimX,
+      %nerltest_print(nerl:string_format("lenDataToRand ~p",[LenDataToRand])),
+      LenData   = rand:uniform(LenDataToRand),
+      LenLabelsToRand = ?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_DIM_Y-?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_MIN_DIM_Y,
+      LenLabels =  rand:uniform(LenLabelsToRand),
+      LenActualData = LenData + ?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_MIN_DIM_X,
+      LenActualLabels = LenLabels + ?NERLWORKER_DISTRIBUTED_FED_WEIGHTED_AVG_CLASSIFIER_DATA_MIN_DIM_Y,
+      io:format("LenActualData ~p~n LenActualLabels ~p~n",[LenActualData, LenActualLabels]),
+
+      if 
+      (LenActualData == LenActualLabels) -> 
+                LenActualDataIf  =  LenActualData+1;
+      true -> 
+                LenActualDataIf  =  LenActualData
+      end,
+      NumOfSamples = 50,
+      DataRand = generate_nerltensor(float,NumOfSamples,LenActualDataIf,1),
+      NerlTensor = DataRand,
+      LayersSizes = nerl:string_format("~p,~p",[LenActualDataIf-LenActualLabels,LenActualLabels]),
+      LayersTypes = "1,3",
+      io:format("before test_nerlworker_nif ~n"),
+      nerlNIF:test_nerlworker_nif(ModelId,ModelType,ModelArgs,LayersSizes, LayersTypes, 
+      LayersFunctionalityCodes, LearningRate, Epochs, OptimizerType, 
+      OptimizerArgs, LossMethod, DistributedSystemType, DistributedSystemArg),
+      io:format("before train_nif ~n"),
+      {NerlTensorDataBinTrain , Type} = nerlNIF:nerltensor_conversion({NerlTensor,erl_float} , float),
+      nerlNIF:train_nif(ModelId , NerlTensorDataBinTrain , Type), 
+      io:format("before get_distributed_system_train_labels_count_nif ~n"),
+      LabelCount = nerlNIF:get_distributed_system_train_labels_count_nif(ModelId),
+      io:format("before remove_nerlworker_nif ~n"),
```

Comment:
@Orisadek 
Remove io:format of debugging please

Moreover, there is no validation of label count, you should count with Erlang the original tensor (DataRand) and compare with LabelCount (after transforming it from bin nerltensor to erl nerltensor)